Sources/NIO/Thread.swift

```diff
@@ -90,51 +128,86 @@ final class NIOThread {
     ///     - name: The name of the `NIOThread` or `nil` if no specific name should be set.
     ///     - body: The function to execute within the spawned `NIOThread`.
     ///     - detach: Whether to detach the thread. If the thread is not detached it must be `join`ed.
-    static func spawnAndRun(name: String? = nil, detachThread: Bool = true, body: @escaping (NIOThread) -> Void) {
-        // Unfortunately the pthread_create method take a different first argument depending on if it's on Linux or macOS, so ensure we use the correct one.
-        #if os(Linux)
-            var pt: pthread_t = pthread_t()
-        #else
-            var pt: pthread_t? = nil
-        #endif
-
-        // Store everything we want to pass into the c function in a Box so we can hand-over the reference.
+    static func spawnAndRun(name: String? = nil, detachThread: Bool = true,
+                            body: @escaping (NIOThread) -> Void) {
+        var handle: NIOThreadHandle = .invalid
+
+        // Store everything we want to pass into the c function in a Box so we
+        // can hand-over the reference.
         let tuple: ThreadBoxValue = (body: body, name: name)
         let box = ThreadBox(tuple)
-        let res = pthread_create(&pt, nil, { p in
-            // Cast to UnsafeMutableRawPointer? and force unwrap to make the same code work on macOS and Linux.
-            let b = Unmanaged<ThreadBox>.fromOpaque((p as UnsafeMutableRawPointer?)!).takeRetainedValue()
 
-            let body = b.value.body
-            let name = b.value.name
+        let argv0 = Unmanaged.passRetained(box).toOpaque()
 
-            let pt = pthread_self()
+#if os(Windows)
+        // FIXME(compnerd) this should use the `stdcall` calling convention
```

Comment:
FIXME left

Reply:
is this for this PR or a followup?

Reply:
It will have to be a follow up.  This requires some new work in the compiler - `stdcall` is not a supported calling convention right now.  This largely is a pedantic concern atm - Windows x86 is broken for other reasons right now.